frontend/src/metabase/static-viz/lib/numbers.ts

```diff
@@ -30,17 +31,33 @@ export const formatNumber = (number: number, options?: NumberFormatOptions) => {
     scale,
     prefix,
     suffix,
+    compact,
   } = { ...DEFAULT_OPTIONS, ...options };
 
-  const format = new Intl.NumberFormat("en", {
-    style: number_style !== "scientific" ? number_style : "decimal",
-    notation: number_style !== "scientific" ? "standard" : "scientific",
-    currency: currency,
-    currencyDisplay: currency_style,
-    useGrouping: true,
-    minimumFractionDigits: decimals,
-    maximumFractionDigits: decimals != null ? decimals : 2,
-  });
+  function createFormat(compact?: boolean) {
```

Comment:
I create this function inside this scope because it uses a lot of local variables, if I were to extract this outside this function it will require a lot of parameters making it less readable imo.